explorer/interpreter/stack_space.h

```diff
@@ -0,0 +1,57 @@
+// Part of the Carbon Language project, under the Apache License v2.0 with LLVM
+// Exceptions. See /LICENSE for license information.
+// SPDX-License-Identifier: Apache-2.0 WITH LLVM-exception
+
+#ifndef CARBON_EXPLORER_INTERPRETER_STACK_SPACE_H_
+#define CARBON_EXPLORER_INTERPRETER_STACK_SPACE_H_
+
+#include <functional>
+#include <optional>
+
+namespace Carbon {
+
+namespace Internal {
```

Comment:
Rather than having an `Internal` namespace, I suggest adding an `// Implementation details only below here` comment, basically demarcating the boundary where we'd split the .h and .cc files if C++ didn't make us put the template code in this header. That way the first thing the reader sees is the public API.

Reply:
I should add -- your suggestion would also require forward declarations for the template functions. I think separating declaration and definition is significant overhead for these short functions, and this header is pretty short to just skim through.